### ISSUE TITLE
Optimize performance

### ISSUE THREAD

**Issue by alfinkel** [cloudant/python-cloudant#67]
We need to consider providing a way for the developer to optimize some of the functionality provided by the library.
- One possible thing to consider is to add a "refresh" local cache attribute to the account, database, and document objects.  This will allow the developer to have more control over the library objects that they use.  Right now account, database, and document extend dict so with every operation performed to an object that updates it remotely raises two compelling questions:
  - Do you automatically refresh the local cached version for consistency/clarity purposes??
  - Do you not automatically refresh the local cached version for performance reasons??
- Right now, each object class (account, database, document) and even each method within the class seem to handle these two questions inconsistently.
- Another consideration that belongs as part of this case and that is somewhat related to the bullet above is the way in which attachments are handled via a document object.  Specifically, when adding or updating attachments we currently perform an extra fetch to refresh the document local dictionary.  It would be beneficial if there was a way to remove this last fetch and instead refresh the local document dictionary based on response content and headers, if possible.
  - See this related PR discussion:  https://github.com/cloudant/python-cloudant/pull/28/files#r44277879

**Comment by robolivable** [comment]
I ran into an issue with the local caching feature of this library when utilizing it with multiple Celery workers to write millions of documents. Within an hour or so of running 5+ workers on a single box, the worker execution times would slow down, as the memory consumption grew drastically (execution times slowed down, I presume due to memory management with swap space).

I was able to work around the issue by making manual detour calls to the database using the internal requests session instance, to bypass the cache. Anyway, the point I'm getting at is that this library can probably also benefit from a "local cache limit" or a "clear local cache" function to provide some more control there.
